Automattic/studio
# Studio: User can't delete expired demo site from Studio

Closes https://github.com/Automattic/dotcom-forge/issues/7082

## Proposed Changes

This PR adds `Clear expired site` button to the expired screen under `Share` tab in Studio:

<img width="1023" alt="Screenshot 2024-06-07 at 4 32 35 PM" src="https://github.com/Automattic/studio/assets/25575134/72ab0f71-286b-4481-9ead-944cdb0ab3a4">

## Testing Instructions

* Pull the changes from this PR 
* Start the app with `nvm use && npm install && npm start`
* Create a demo site
* Force the demo site to be expired by changing the [following line](https://github.com/Automattic/studio/blob/ffd73b9a60113ed55785ea03fed6aeaf999d937f/src/hooks/use-expiration-date.ts#L24) in the `use-expiration-date.ts`:

```suggestion
let isExpired = false;
```

to `true`:

```suggestion
let isExpired = true
```

* Observe `Clear expired site` button
* Click the button, observe a spinner and confirm that you are seeing the `<NoSnapshots>` component
* Try testing `Clear expired site` without internet connection and confirm that the button is disabled in this case

**Notes**

* There is a loading spinner present which was not mentioned in the initial design. I tried without it but I found the shift in the layout to be too abrupt. Happy to reconsider this approach
* Once you focus on the button to `Clear expired site`, you will see the borders appear. I am not sure if they should be there according to the design. Happy to remove those as well.

## Pre-merge Checklist
